aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -694,6 +695,53 @@ Tensor new_empty_batching_rule(
   return physical_view.newLogicalFromPhysical(result);
 }
 
+Tensor new_empty_strided_batching_rule(
+    const Tensor& self,
+    IntArrayRef size,
+    IntArrayRef stride,
+    const TensorOptions& options) {
+  auto physical_view = MultiBatchVmapTransform::logicalToPhysical(self);
+  auto physical_size = physical_view.getPhysicalShape(size);
+
+  // Let [B0, B1, B2] be the shape of the batch dims. We know what the physical
+  // shape of the result should be ([B0, B1, B2] + size), but what about
+  // the physical strides?
+  //
+  // We're actually free to pick whatever stride we want:
+  // e.g., for size=[5, 3], stride=[0, 1], we could decide to
+  // use
+  // - physical size: [B0, B1, B2, 5, 3]
+  // - physical stride: [100*B1*B2, 100*B2, 100, 0, 1]
+  //
+  // Let's select the strides that result in the smallest memory usage.
+  // Let S be the size of the storage if one were to construct a tensor
+  // with `size` and `stride` via empty_strided(size, stride).
+  // Then the physical sizes/strides should be:
+  // - physical size: [B0, B1, B2, 5, 3]
+  // - physical stride: [B1 * B2 * S, B2 * S, S, 0, 1]
+  auto batch_shape = IntArrayRef(
+      physical_view.tensor().sizes().begin(), physical_view.numBatchDims());
+
+  // physical_strides = [B1 * B2 * S, B2 * S, S]
+  auto physical_strides = at::detail::defaultStrides(batch_shape);
+  TORCH_CHECK(size.size() == stride.size(),
+        "new_empty_strided(sizes, strides): dimensionality of sizes (",
+        size.size(), ") must match dimensionality of strides (",
+        stride.size(), ")");
+  auto example_storage_size = native::storage_size_for(size, stride);
+  for (int64_t idx = 0; idx < physical_strides.size(); ++idx) {
+    physical_strides[idx] *= example_storage_size;
+  }
+
+  // physical_strides = [B1 * B2 * S, B2 * S, S] + strides
+  physical_strides.insert(physical_strides.end(), stride.begin(), stride.end());
+
+  auto result = physical_view.tensor().new_empty_strided(
```

Comment:
nit: Do you have to call back into `new_empty_strided` here? Or could it be done with `at::empty_strided` and this is just for consistency?

Reply:
This can be done with at::empty_strided. I chose to use `new_empty_strided` because it was the first thing that came to mind. I like at::empty_strided because it's less verbose, let's go with that.

Reply:
Actually, I take that back: there is indeed a semantic difference. If the user didn't provide some of the options, then we use the options of the original tensor. For example:

```
x = torch.rand(3, device='cuda')
y = x.new_empty_strided([4, 4], [4, 1])
```
this should create a new CUDA tensor. However, none of the options being passed to `new_empty_strided_batching_rule` have actually been set. So if we used `at::empty(physical_size, physical_strides, options)`, this would end up creating a CPU tensor (the default for device when it is not set is CPU)